scripts/quests/otherAreas/Under_the_Sea.lua

```diff
@@ -0,0 +1,135 @@
+-----------------------------------
+-- Under the Sea
+-----------------------------------
+-- Log ID: 4, Quest ID: 17
+-- Yaya    : !pos -18.770 -2.597 -14.929 248
+-- Oswald  : !pos 47.119 -15.273 7.989 248
+-- Jimaida : !pos -17.342 -2.597 -18.766 248
+-- Zaldon  : !pos -11.810 -7.287 -6.742
+-----------------------------------
+
+local quest = Quest:new(xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.UNDER_THE_SEA)
+
+quest.reward =
+{
+    item  = xi.item.AMBER_EARRING,
+    title = xi.title.LIL_CUPID,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
```

Comment:
While we have the fishing setting set as `    FISHING_ENABLE = false,` by default, you should probably check that here too to make sure people can't start the fishing questlines if fishing is globally off.

Reply:
While you're here, can you change and audit how this setting is used?
```lua
    -- Enables fishing. 0 = Disabled. 1 = Enable. ENABLE AT YOUR OWN RISK.
    FISHING_ENABLE = false,
```

I say this because one of Lua's gotchas is that 0 and 1 are both truth-y. The only things that are false-y are `nil` and `false`, so it's very easy to fall into the trap here of mixing 0/1 and true/false in the settings and not getting consistent behaviour.

So in the check in this quest, check specifically for `== 1` or `== true`, and then make sure the setting itself and the associated comments reflect whether or not you're using 0/1 or booleans. We're not particularly consistent about which we use across the board, so pick whichever you prefer, or whichever seems most prevalent.

Reply:
(and update other areas that look at that setting to be the same)

Reply:
It looks like true/false seems to be the norm, for settings at least.  Only other place I see it used is in the packet_system.cpp and fishing_util.cpp.